openhands/runtime/client/runtime.py

```diff
@@ -317,7 +317,7 @@ def close(self, close_client: bool = True):
         containers = self.docker_client.containers.list(all=True)
         for container in containers:
             try:
-                if container.name.startswith(self.container_name_prefix):
+                if container.name == self.container_name:
```

Comment:
I had to change this otherwise because the tests are running in parallel, it would get and delete wrong containers.
I'm not sure of the full impact of this particular change so would like someone in the know to confirm it makes sense. Logically it makes sense.

Reply:
Could you run these runtime tests locally and observe in Docker desktop whether the containers are removed correctly during the tests, still?

Reply:
Yep they are getting removed correctly.

Reply:
Maybe better to gate this with an env var that we set in the tests.

The main problem is that if the app doesn't shut down properly, it'll leave runtimes sitting around on the user's machine. This cleans them up the next time it runs

Reply:
Ohhh....I see....that's unfortunate :D

Reply:
Robert, I'd like to merge this because it works consistently at least with two CPUs and saves about 16min. I'm wondering how do I go about doing this?

Thinking having an environment variable might be weird if it's just for tests. What about adding a param to the close() method:
def close(self, close_client: bool = True, rm_all_containers: bool = True)

Then in the runtime tests, set `rm_all_containers` to False and the associated logic

Reply:
That sounds good to me!